Frameworks/UIKit/UIViewController.mm

```diff
@@ -1250,18 +1256,29 @@ - (void)dismissViewControllerAnimated:(BOOL)animated completion:(void (^)(void))
         curController->priv->_parentViewController->priv->_modalViewController = nil;
     }
 
-    curController->priv->_parentViewController = nil;
-    curController->priv->_presentingViewController = nil;
-    curController->priv->_popoverPresentationController = nil;
-
     UIView* curView = [curController view];
 
     UIView* myView = [self view];
     [myView setHidden:FALSE];
 
     if (realPopoverPresented) {
-        [[curController popoverPresentationController] _dismissAnimated:animated completion:completion];
-    } else if (animated) {
+        __block __unsafe_unretained UIViewController* weakCurController = curController;
+
+        [[curController popoverPresentationController] _dismissAnimated:animated completion:^{
+            StrongId<UIViewController> strongCurController = weakCurController;
+            [strongCurController _onDismissCleanup];
+
+            if (completion) {
+                completion();
+            }
+        }];
+
+        return;
+    }
+
+    [curController _onDismissCleanup];
```

Comment:
>[curController _onDismissCleanup]; [](start = 4, length = 34)

should this be moved to end of the function? because we still see curController being used below.